packages/pwa-kit-dev/bin/pwa-kit-dev.js

```diff
@@ -86,6 +86,50 @@ const validateAppConfiguration = () => {
     }
 }
 
+/**
+ * Executes a command in the application-extension directory.
+ * @param {string} command - The command to execute.
+ * @param {string} pkgRoot - The root directory of the package.
+ */
+const runCommandInExtensions = (command, pkgRoot) => {
+    const extensionsDir = p.join(pkgRoot, '..', '..', '..', 'app', 'application-extensions')
+
+    if (fse.existsSync(extensionsDir)) {
+        const extensions = fse.readdirSync(extensionsDir).filter((dir) => {
+            const dirPath = p.join(extensionsDir, dir)
+            return fse.statSync(dirPath).isDirectory()
+        })
+
+        extensions.forEach((extension) => {
+            const extensionPath = p.join(extensionsDir, extension)
+
+            // If the extension is namespaced, look for the next directory
+            if (extension.startsWith('@')) {
+                const subDirs = fse.readdirSync(extensionPath).filter((subDir) => {
+                    return fse.statSync(p.join(extensionPath, subDir)).isDirectory()
+                })
+
+                // Execute the provided command in each subdirectory for namespaced extensions
+                subDirs.forEach((subDir) => {
+                    const subDirPath = p.join(extensionPath, subDir)
+                    try {
+                        execSync(`cd ${subDirPath} && ${command}`)
+                    } catch (error) {
+                        console.error(`Error executing command in ${subDirPath}:`, error.message)
+                    }
+                })
+            } else {
+                // For non-namespaced extensions, execute the provided command in the extension directory
+                try {
+                    execSync(`cd ${extensionPath} && ${command}`)
+                } catch (error) {
+                    console.error(`Error executing command in ${extensionPath}:`, error.message)
+                }
+            }
+        })
+    }
```

Comment:
The contents of this loop is quite big and have some repetition (like the execSync).. try doing something like getting a list of all the directories of extensions (namespaced and non-namespaced) .. then iterate though that list. It should shorten things up and make it easier to read.